mecanum_drive_controller:
  ros__parameters:
    wheel_names: ["wheel0","wheel1","wheel2","wheel3"]
    write_op_modes: ["motor_controller"]

    wheel_separation_x: 0.40
    wheel_separation_y: 0.40
    wheels_per_side: 1  # actually 2, but both are controlled by 1 signal <- unsure
    wheel_radius: 0.02

    wheel_separation_multiplier: 1.0
    left_wheel_radius_multiplier: 1.0
    right_wheel_radius_multiplier: 1.0

    odom_frame_id: odom
    base_frame_id: base_link
    pose_covariance_diagonal: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    twist_covariance_diagonal: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]

    open_loop: true
    enable_odom_tf: true

    cmd_vel_timeout: 500 # milliseconds
    velocity_rolling_window_size: 10

    linear.x.has_velocity_limits: false
    linear.x.has_acceleration_limits: false
    linear.x.max_velocity: 0.0
    linear.x.min_velocity: 0.0
    linear.x.max_acceleration: 0.0

    linear.y.has_velocity_limits: false
    linear.y.has_acceleration_limits: false
    linear.y.max_velocity: 0.0
    linear.y.min_velocity: 0.0
    linear.y.max_acceleration: 0.0

    angular.z.has_velocity_limits: false
    angular.z.has_acceleration_limits: false
    angular.z.max_velocity: 0.0
    angular.z.min_velocity: 0.0
    angular.z.max_acceleration: 0.0
    angular.z.min_acceleration: 0.0
